### PR TITLE
Fix the 'clean_affiliation' function in register_cme form

### PR DESCRIPTION
**Pull request by caesar2164**
@dcadams this doesn't completely fix this, I'll show you the problem tomorrow...

CC @caseylitton

### CODE REVIEW

File: openedx/stanford/djangoapps/register_cme/forms.py [hunk 1]
```diff
@@ -139,22 +146,23 @@ def clean_affiliation(self):
...
-                    raise forms.ValidationError(
-                        value,
-                        code='invalid',
-                    )
+        errors = []
+        sunet_id = self.data['sunet_id']
+        stanford_department = self.data['stanford_department']
+        if not(sunet_id):
+            errors.append({'sunet_id', 'required'})
+        if (2 < sunet_id < 9):
```
Comment (dcadams) [inline openedx/stanford/djangoapps/register_cme/forms.py]: elif

File: openedx/stanford/djangoapps/register_cme/forms.py [hunk 2]
```diff
@@ -139,22 +146,23 @@ def clean_affiliation(self):
         affiliation = self.cleaned_data['affiliation']
```
Comment (dcadams) [inline openedx/stanford/djangoapps/register_cme/forms.py]: This method clearly didn't have a unit test. Not necessarily with this fix but can you add one shortly thereafter.

Comment (caesar2164) [inline openedx/stanford/djangoapps/register_cme/forms.py]: sounds good

File: openedx/stanford/djangoapps/register_cme/forms.py [hunk 3]
```diff
@@ -139,22 +146,23 @@ def clean_affiliation(self):
...
-                    raise forms.ValidationError(
-                        value,
-                        code='invalid',
-                    )
+        errors = []
+        sunet_id = self.data['sunet_id']
+        stanford_department = self.data['stanford_department']
+        if not(sunet_id):
+            errors.append({'sunet_id', 'required'})
+        elif (2 < sunet_id < 9):
```
Comment (caseylitton) [inline openedx/stanford/djangoapps/register_cme/forms.py]: This looks like it should check len(sunet_id).  But then the condition would be throwing an error if the length is between 2 and 9.  Which is opposite of what we want.

File: openedx/stanford/djangoapps/register_cme/forms.py [hunk 4]
```diff
@@ -139,22 +146,23 @@ def clean_affiliation(self):
...
-                    raise forms.ValidationError(
-                        value,
-                        code='invalid',
-                    )
+        errors = []
+        sunet_id = self.data['sunet_id']
+        stanford_department = self.data['stanford_department']
+        if not(sunet_id):
+            errors.append({'sunet_id', 'required'})
+        elif (len(sunet_id) < 3 or len(sunet_id) > 9):
```
Comment (dcadams) [inline openedx/stanford/djangoapps/register_cme/forms.py]: elif len(sunet_id) < 3 or len(sunet_id) > 9:

**Comment by caseylitton** [comment]
+1

File: openedx/stanford/djangoapps/register_cme/forms.py [hunk 5]
```diff
@@ -139,22 +146,23 @@ def clean_affiliation(self):
...
-                    raise forms.ValidationError(
-                        value,
-                        code='invalid',
-                    )
+        errors = []
+        sunet_id = self.data['sunet_id']
+        stanford_department = self.data['stanford_department']
+        if not(sunet_id):
+            errors.append({'sunet_id', 'required'})
+        elif len(sunet_id) < 3 or len(sunet_id) > 9:
```
Comment (dcadams) [inline openedx/stanford/djangoapps/register_cme/forms.py]: There's an inconsistency now between the 'invalid' message and this. (between the lengths of).

**Pull request merged**